### ISSUE TITLE
Possible version mismatch on exercise "Anagram"

### ISSUE THREAD

**Issue by Pedro-V** [exercism/scheme-test-runner#37]
Hey, I have recently completed the [anagram exercise](https://exercism.org/tracks/scheme/exercises/anagram). It works perfectly on my machine: 

    $ make chez
    scheme --script test.scm
    
    Well done!

But in the exercise's online test, It shows me an ambiguous message:

> An error occurred while running your tests. This might mean that there was an issue in our infrastructure, or it might mean that you have something in your code that's causing our systems to break.
> 
> Please check your code, and if nothing seems to be wrong, try running the tests again.

I really don't know how to proceed. The solution seems perfect to me. It's located on one of my public repos:

[my solution file](https://raw.githubusercontent.com/Pedro-V/scheme-ideas-exercises/main/exercism/anagram.scm)

### Some info on my environment

* `scheme --version` = `9.5`
* `make --version` = `GNU Make 4.2.1 Built for x86_64-pc-linux-gnu`
* `uname -a` = `Linux note 5.10.102.1-microsoft-standard-WSL2 #1 SMP Wed Mar 2 00:30:59 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux`
* `lsb_release -d` =  `Ubuntu 20.04.4 LTS`

**Comment by guygastineau** [comment]
Thank you for reporting this.

Maybe @ErikSchierboom or @iHiD could get the `stderr` output of the container for us to inspect.  In the meantime, can you verify that your exercise tests are the latest version, @Pedro-V ?

**Comment by Pedro-V** [comment]
> verify that your exercise test are the latest version

Hey @guygastineau, thanks for taking the time for helping me.

I'm not sure how I would verify that the exercise test is the latest version.
I tried doing `exercism version`. If that's the correct way, then the output is:

    exercism version 3.0.13

**Comment by guygastineau** [comment]
In the website, if you go to the exercise page for anagram on the scheme track it should have a button to upgrade to newest version of the exercise if one is available.  Still, this shouldn't be a problem even with tests from a few versions ago.

Hopefully we can get the stderr logs from the container soon.

---

This will help a little though, do you have a file called `test-util.ss` in your exercise?  We recently changed tests to use this file and simplify `test.scm`.

**Comment by Pedro-V** [comment]
Yes, I do have `test-util.ss`.

**Comment by ErikSchierboom** [comment]
@guygastineau This is the stderr output:

```console
"Exception in read: invalid sharp-sign prefix #< on #<input port pid 9 stdout>\n"
```

**Comment by guygastineau** [comment]
That looks like somehow the test serialized an object that has no textual isomorphism!  More specifically, It looks like ~stdout was serialized~ a procedure was serialized in exception handling!

@Pedro-V sorry I haven't had more time for this over the last two weeks.  Please run the following and post the output here :)

```
$ chez --script test.scm --docker
```

The following would also be good to see, since the tests are run in both chez and guile (although we should be catching errors and choosing the one that works out best):
```
$ guile test.scm --docker
```

Thank you.

**Comment by Pedro-V** [comment]
> @Pedro-V sorry I haven't had more time for this over the last two weeks. 

That's ok! Just taking the time to help is a really nice act.

Here's the output of both commands:

**ChezScheme**
```scheme
((test-lib-version . 1) (status . ok) (pass (description . "no matches") (code test-success "no matches" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("diaper" ("hello" "world" "zombies" "pants"))) (quote ())) (stdout . "")) (pass (description . "detects two anagrams") (code test-success "detects two anagrams" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("master" ("stream" "pigeon" "maters"))) (quote ("stream" "maters"))) (stdout . "")) (pass (description . "does not detect anagram subsets") (code test-success "does not detect anagram subsets" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("good" ("dog" "goody"))) (quote ())) (stdout . "")) (pass (description . "detects anagram") (code test-success "detects anagram" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("listen" ("enlists" "google" "inlets" "banana"))) (quote ("inlets"))) (stdout . "")) (pass (description . "detects three anagrams") (code test-success "detects three anagrams" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("allergy" ("gallery" "ballerina" "regally" "clergy" "largely" "leading"))) (quote ("gallery" "regally" "largely"))) (stdout . "")) (pass (description . "detects multiple anagrams with different case") (code test-success "detects multiple anagrams with different case" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("nose" ("Eons" "ONES"))) (quote ("Eons" "ONES"))) (stdout . "")) (pass (description . "does not detect non-anagrams with identical checksum") (code test-success "does not detect non-anagrams with identical checksum" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("mass" ("last"))) (quote ())) (stdout . "")) (pass (description . "detects anagrams case-insensitively") (code test-success "detects anagrams case-insensitively" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("Orchestra" ("cashregister" "Carthorse" "radishes"))) (quote ("Carthorse"))) (stdout . "")) (pass (description . "detects anagrams using case-insensitive subject") (code test-success "detects anagrams using case-insensitive subject" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("Orchestra" ("cashregister" "carthorse" "radishes"))) (quote ("carthorse"))) (stdout . "")) (pass (description . "detects anagrams using case-insensitive possible matches") (code test-success "detects anagrams using case-insensitive possible matches" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("orchestra" ("cashregister" "Carthorse" "radishes"))) (quote ("Carthorse"))) (stdout . "")) (pass (description . "does not detect an anagram if the original word is repeated") (code test-success "does not detect an anagram if the original word is repeated" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("go" ("go Go GO"))) (quote ())) (stdout . "")) (pass (description . "anagrams must use all letters exactly once") (code test-success "anagrams must use all letters exactly once" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("tapper" ("patter"))) (quote ())) (stdout . "")) (pass (description . "words are not anagrams of themselves (case-insensitive)") (code test-success "words are not anagrams of themselves (case-insensitive)" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("BANANA" ("BANANA" "Banana" "banana"))) (quote ())) (stdout . "")) (pass (description . "words other than themselves can be anagrams") (code test-success "words other than themselves can be anagrams" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("LISTEN" ("Listen" "Silent" "LISTEN"))) (quote ("Silent"))) (stdout . "")))
```


**Guile**
```scheme
WARNING: (guile-user): imported module (rnrs) overrides core binding `write'
WARNING: (guile-user): imported module (rnrs) overrides core binding `map'
WARNING: (guile-user): imported module (rnrs) overrides core binding `close-output-port'
((test-lib-version . 1) (status . ok) (fail (description . "no matches") (code test-success "no matches" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("diaper" ("hello" "world" "zombies" "pants"))) (quote ())) (input "diaper" ("hello" "world" "zombies" "pants")) (expected) (actual error "sort" "Wrong type argument in position ~A: ~S" 1 #<procedure char<? (#:optional _ _ . _)>) (stdout . "")) (fail (description . "detects two anagrams") (code test-success "detects two anagrams" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("master" ("stream" "pigeon" "maters"))) (quote ("stream" "maters"))) (input "master" ("stream" "pigeon" "maters")) (expected "stream" "maters") (actual error "sort" "Wrong type argument in position ~A: ~S" 1 #<procedure char<? (#:optional _ _ . _)>) (stdout . "")) (fail (description . "does not detect anagram subsets") (code test-success "does not detect anagram subsets" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("good" ("dog" "goody"))) (quote ())) (input "good" ("dog" "goody")) (expected) (actual error "sort" "Wrong type argument in position ~A: ~S" 1 #<procedure char<? (#:optional _ _ . _)>) (stdout . "")) (fail (description . "detects anagram") (code test-success "detects anagram" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("listen" ("enlists" "google" "inlets" "banana"))) (quote ("inlets"))) (input "listen" ("enlists" "google" "inlets" "banana")) (expected "inlets") (actual error "sort" "Wrong type argument in position ~A: ~S" 1 #<procedure char<? (#:optional _ _ . _)>) (stdout . "")) (fail (description . "detects three anagrams") (code test-success "detects three anagrams" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("allergy" ("gallery" "ballerina" "regally" "clergy" "largely" "leading"))) (quote ("gallery" "regally" "largely"))) (input "allergy" ("gallery" "ballerina" "regally" "clergy" "largely" "leading")) (expected "gallery" "regally" "largely") (actual error "sort" "Wrong type argument in position ~A: ~S" 1 #<procedure char<? (#:optional _ _ . _)>) (stdout . "")) (fail (description . "detects multiple anagrams with different case") (code test-success "detects multiple anagrams with different case" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("nose" ("Eons" "ONES"))) (quote ("Eons" "ONES"))) (input "nose" ("Eons" "ONES")) (expected "Eons" "ONES") (actual error "sort" "Wrong type argument in position ~A: ~S" 1 #<procedure char<? (#:optional _ _ . _)>) (stdout . "")) (fail (description . "does not detect non-anagrams with identical checksum") (code test-success "does not detect non-anagrams with identical checksum" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("mass" ("last"))) (quote ())) (input "mass" ("last")) (expected) (actual error "sort" "Wrong type argument in position ~A: ~S" 1 #<procedure char<? (#:optional _ _ . _)>) (stdout . "")) (fail (description . "detects anagrams case-insensitively") (code test-success "detects anagrams case-insensitively" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("Orchestra" ("cashregister" "Carthorse" "radishes"))) (quote ("Carthorse"))) (input "Orchestra" ("cashregister" "Carthorse" "radishes")) (expected "Carthorse") (actual error "sort" "Wrong type argument in position ~A: ~S" 1 #<procedure char<? (#:optional _ _ . _)>) (stdout . "")) (fail (description . "detects anagrams using case-insensitive subject") (code test-success "detects anagrams using case-insensitive subject" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("Orchestra" ("cashregister" "carthorse" "radishes"))) (quote ("carthorse"))) (input "Orchestra" ("cashregister" "carthorse" "radishes")) (expected "carthorse") (actual error "sort" "Wrong type argument in position ~A: ~S" 1 #<procedure char<? (#:optional _ _ . _)>) (stdout . "")) (fail (description . "detects anagrams using case-insensitive possible matches") (code test-success "detects anagrams using case-insensitive possible matches" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("orchestra" ("cashregister" "Carthorse" "radishes"))) (quote ("Carthorse"))) (input "orchestra" ("cashregister" "Carthorse" "radishes")) (expected "Carthorse") (actual error "sort" "Wrong type argument in position ~A: ~S" 1 #<procedure char<? (#:optional _ _ . _)>) (stdout . "")) (fail (description . "does not detect an anagram if the original word is repeated") (code test-success "does not detect an anagram if the original word is repeated" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("go" ("go Go GO"))) (quote ())) (input "go" ("go Go GO")) (expected) (actual error "sort" "Wrong type argument in position ~A: ~S" 1 #<procedure char<? (#:optional _ _ . _)>) (stdout . "")) (fail (description . "anagrams must use all letters exactly once") (code test-success "anagrams must use all letters exactly once" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("tapper" ("patter"))) (quote ())) (input "tapper" ("patter")) (expected) (actual error "sort" "Wrong type argument in position ~A: ~S" 1 #<procedure char<? (#:optional _ _ . _)>) (stdout . "")) (fail (description . "words are not anagrams of themselves (case-insensitive)") (code test-success "words are not anagrams of themselves (case-insensitive)" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("BANANA" ("BANANA" "Banana" "banana"))) (quote ())) (input "BANANA" ("BANANA" "Banana" "banana")) (expected) (actual error "sort" "Wrong type argument in position ~A: ~S" 1 #<procedure char<? (#:optional _ _ . _)>) (stdout . "")) (fail (description . "words other than themselves can be anagrams") (code test-success "words other than themselves can be anagrams" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("LISTEN" ("Listen" "Silent" "LISTEN"))) (quote ("Silent"))) (input "LISTEN" ("Listen" "Silent" "LISTEN")) (expected "Silent") (actual error "sort" "Wrong type argument in position ~A: ~S" 1 #<procedure char<? (#:optional _ _ . _)>) (stdout . "")))
```

**Comment by guygastineau** [comment]
> #<procedure char<? (#:optional _ _ . _)>

Ah, as I thought, an non-deserializable object was serialized in the docker output of the test.  I did not consider this when adding better error reporting to the tests.  I guess I will have to switch the test utility code to convert error lists into strings before serialization to avoid this issue.  Silly me.  Here I was thinking `(read (write x)) = x`, but clearly I've spent too much time with typed lambda calculus.  I would do well to remember that scheme embraces Russel's Paradox.

**Comment by guygastineau** [comment]
I will work on a patch to the `test-util.scm` code in the scheme repo, and I will update those files here.  I will also include your solution as a test here to guard against regressions for this issue.

**Comment by guygastineau** [comment]
~Oh man, this is so confusing.  Running it with Guile on my local machine I don't get the error at all~  I was using a very old version of the test that I pulled from my solution to anagram.  Please excuse any alarm I may have caused with the above.

**Comment by guygastineau** [comment]
I am using guile 2.2.7.   Still, your error has highlighted a bug in the runner, so I will work out a test that catches this.

**Comment by guygastineau** [comment]
@ErikSchierboom I believe I have this fixed.  Once again we need to merge a PR here and in the regular scheme track repo.  Given there is no portable scheme test faciliy, we seem stuck with the burden of keeping some parts in sync ourselves.

@Pedro-V Once exercism/scheme#266 is merged, you should be able to upgrade your exercise through the exercism website UI, and then your solution should pass the online tests.  Unfortunately, I think you will need to change at least one byte in your solution in order to get the runner to test it.  Maybe just try changing a comment or something since your solution works.  Anyway, you will still have to wait for the merges before then.

**Comment by Pedro-V** [comment]
> Unfortunately, I think you will need to change at least one byte in your solution in order to get the runner to test it. 

Very well, glad u got to fix the problem that quickly!

I'm really thankful for the help. I will close this issue once the PR is merged and I can test the new version.

**Comment by guygastineau** [comment]
Thank you for your kind words.  I added a line in the PR that closes this issue when it gets merged.  That wouldn't leave time for you to test things though.  I can remove it, or you can reopen if it gets auto closed but your tests still fail.  Let me know which you prefer.  I don't know when we will get the reviews necessary for merging.  Probably admins will have some time on Monday or Tuesday.

I hope my commit messages aren't too scatter brained.  I had covid this last week, so I was doing the work with a fever 🤣

**Comment by ErikSchierboom** [comment]
@guygastineau I've approved both PR.

Sorry to hear about the covid! Hope you're okay.

**Comment by guygastineau** [comment]
Thank you, Erik.

@Pedro-V the fix has been merged in both repositories.  It might take a few minutes for the exercism UI to let you upgrade your exercise.  Please try at your earliest convenience.  Even though we are using your code in a regression test now, I still removed the closing effect on this issue from the issue, so please close this issue yourself when you are satisfied your issue is solved.

**Comment by Pedro-V** [comment]
Just ran the tests again. The input code is the one that opened this issue, but I altered some bytes and reverted. The error is the same:

> An error occurred while running your tests. This might mean that there was an issue in our infrastructure, or it might mean that you have something in your code that's causing our systems to break.
> 
> Please check your code, and if nothing seems to be wrong, try running the tests again.

So I ran both docker checks:
```
$ scheme --script test.scm --docker

((test-lib-version . 1) (status . ok) (pass (description . "no matches") (code test-success " matches" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (ote ("diaper" ("hello" "world" "zombies" "pants"))) (quote ())) (stdout . "")) (pass (descripon . "detects two anagrams") (code test-success "detects two anagrams" (lambda (xs ys) (equal(list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("master" ("stream" "pigeon"maters"))) (quote ("stream" "maters"))) (stdout . "")) (pass (description . "does not detect agram subsets") (code test-success "does not detect anagram subsets" (lambda (xs ys) (equal? ist-sort string<? xs) (list-sort string<? ys))) anagram (quote ("good" ("dog" "goody"))) (quo ())) (stdout . "")) (pass (description . "detects anagram") (code test-success "detects anagm" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote (isten" ("enlists" "google" "inlets" "banana"))) (quote ("inlets"))) (stdout . "")) (pass (desiption . "detects three anagrams") (code test-success "detects three anagrams" (lambda (xs ys(equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("allergy" ("gallery"ballerina" "regally" "clergy" "largely" "leading"))) (quote ("gallery" "regally" "largely"))(stdout . "")) (pass (description . "detects multiple anagrams with different case") (code te-success "detects multiple anagrams with different case" (lambda (xs ys) (equal? (list-sort sing<? xs) (list-sort string<? ys))) anagram (quote ("nose" ("Eons" "ONES"))) (quote ("Eons" "ES"))) (stdout . "")) (pass (description . "does not detect non-anagrams with identical checkm") (code test-success "does not detect non-anagrams with identical checksum" (lambda (xs ys)equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("mass" ("last"))) (qte ())) (stdout . "")) (pass (description . "detects anagrams case-insensitively") (code testuccess "detects anagrams case-insensitively" (lambda (xs ys) (equal? (list-sort string<? xs) ist-sort string<? ys))) anagram (quote ("Orchestra" ("cashregister" "Carthorse" "radishes")))quote ("Carthorse"))) (stdout . "")) (pass (description . "detects anagrams using case-insensive subject") (code test-success "detects anagrams using case-insensitive subject" (lambda (xys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("Orchestra" ("chregister" "carthorse" "radishes"))) (quote ("carthorse"))) (stdout . "")) (pass (description "detects anagrams using case-insensitive possible matches") (code test-success "detects anagms using case-insensitive possible matches" (lambda (xs ys) (equal? (list-sort string<? xs) (st-sort string<? ys))) anagram (quote ("orchestra" ("cashregister" "Carthorse" "radishes"))) uote ("Carthorse"))) (stdout . "")) (pass (description . "does not detect an anagram if the oginal word is repeated") (code test-success "does not detect an anagram if the original word  repeated" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram uote ("go" ("go Go GO"))) (quote ())) (stdout . "")) (pass (description . "anagrams must use l letters exactly once") (code test-success "anagrams must use all letters exactly once" (lama (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("tapper" patter"))) (quote ())) (stdout . "")) (pass (description . "words are not anagrams of themsels (case-insensitive)") (code test-success "words are not anagrams of themselves (case-insensive)" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote"BANANA" ("BANANA" "Banana" "banana"))) (quote ())) (stdout . "")) (pass (description . "wordother than themselves can be anagrams") (code test-success "words other than themselves can banagrams" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (ote ("LISTEN" ("Listen" "Silent" "LISTEN"))) (quote ("Silent"))) (stdout . "")))
```

And with guile:

```` 
$ guile test.scm --docker
WARNING: (guile-user): imported module (rnrs) overrides core binding `write'
WARNING: (guile-user): imported module (rnrs) overrides core binding `map'
WARNING: (guile-user): imported module (rnrs) overrides core binding `close-output-port'
((test-lib-version . 1) (status . ok) (fail (description . "no matches") (code test-success "no matches" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("diaper" ("hello" "world" "zombies" "pants"))) (quote ())) (input "diaper" ("hello" "world" "zombies" "pants")) (expected) (actual error "sort" "Wrong type argument in position ~A: ~S" 1 #<procedure char<? (#:optional _ _ . _)>) (stdout . "")) (fail (description . "detects two anagrams") (code test-success "detects two anagrams" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("master" ("stream" "pigeon" "maters"))) (quote ("stream" "maters"))) (input "master" ("stream" "pigeon" "maters")) (expected "stream" "maters") (actual error "sort" "Wrong type argument in position ~A: ~S" 1 #<procedure char<? (#:optional _ _ . _)>) (stdout . "")) (fail (description . "does not detect anagram subsets") (code test-success "does not detect anagram subsets" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("good" ("dog" "goody"))) (quote ())) (input "good" ("dog" "goody")) (expected) (actual error "sort" "Wrong type argument in position ~A: ~S" 1 #<procedure char<? (#:optional _ _ . _)>) (stdout . "")) (fail (description . "detects anagram") (code test-success "detects anagram" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("listen" ("enlists" "google" "inlets" "banana"))) (quote ("inlets"))) (input "listen" ("enlists" "google" "inlets" "banana")) (expected "inlets") (actual error "sort" "Wrong type argument in position ~A: ~S" 1 #<procedure char<? (#:optional _ _ . _)>) (stdout . "")) (fail (description . "detects three anagrams") (code test-success "detects three anagrams" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("allergy" ("gallery" "ballerina" "regally" "clergy" "largely" "leading"))) (quote ("gallery" "regally" "largely"))) (input "allergy" ("gallery" "ballerina" "regally" "clergy" "largely" "leading")) (expected "gallery" "regally" "largely") (actual error "sort" "Wrong type argument in position ~A: ~S" 1 #<procedure char<? (#:optional _ _ . _)>) (stdout . "")) (fail (description . "detects multiple anagrams with different case") (code test-success "detects multiple anagrams with different case" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("nose" ("Eons" "ONES"))) (quote ("Eons" "ONES"))) (input "nose" ("Eons" "ONES")) (expected "Eons" "ONES") (actual error "sort" "Wrong type argument in position ~A: ~S" 1 #<procedure char<? (#:optional _ _ . _)>) (stdout . "")) (fail (description . "does not detect non-anagrams with identical checksum") (code test-success "does not detect non-anagrams with identical checksum" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("mass" ("last"))) (quote ())) (input "mass" ("last")) (expected) (actual error "sort" "Wrong type argument in position ~A: ~S" 1 #<procedure char<? (#:optional _ _ . _)>) (stdout . "")) (fail (description . "detects anagrams case-insensitively") (code test-success "detects anagrams case-insensitively" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("Orchestra" ("cashregister" "Carthorse" "radishes"))) (quote ("Carthorse"))) (input "Orchestra" ("cashregister" "Carthorse" "radishes")) (expected "Carthorse") (actual error "sort" "Wrong type argument in position ~A: ~S" 1 #<procedure char<? (#:optional _ _ . _)>) (stdout . "")) (fail (description . "detects anagrams using case-insensitive subject") (code test-success "detects anagrams using case-insensitive subject" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("Orchestra" ("cashregister" "carthorse" "radishes"))) (quote ("carthorse"))) (input "Orchestra" ("cashregister" "carthorse" "radishes")) (expected "carthorse") (actual error "sort" "Wrong type argument in position ~A: ~S" 1 #<procedure char<? (#:optional _ _ . _)>) (stdout . "")) (fail (description . "detects anagrams using case-insensitive possible matches") (code test-success "detects anagrams using case-insensitive possible matches" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("orchestra" ("cashregister" "Carthorse" "radishes"))) (quote ("Carthorse"))) (input "orchestra" ("cashregister" "Carthorse" "radishes")) (expected "Carthorse") (actual error "sort" "Wrong type argument in position ~A: ~S" 1 #<procedure char<? (#:optional _ _ . _)>) (stdout . "")) (fail (description . "does not detect an anagram if the original word is repeated") (code test-success "does not detect an anagram if the original word is repeated" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("go" ("go Go GO"))) (quote ())) (input "go" ("go Go GO")) (expected) (actual error "sort" "Wrong type argument in position ~A: ~S" 1 #<procedure char<? (#:optional _ _ . _)>) (stdout . "")) (fail (description . "anagrams must use all letters exactly once") (code test-success "anagrams must use all letters exactly once" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("tapper" ("patter"))) (quote ())) (input "tapper" ("patter")) (expected) (actual error "sort" "Wrong type argument in position ~A: ~S" 1 #<procedure char<? (#:optional _ _ . _)>) (stdout . "")) (fail (description . "words are not anagrams of themselves (case-insensitive)") (code test-success "words are not anagrams of themselves (case-insensitive)" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("BANANA" ("BANANA" "Banana" "banana"))) (quote ())) (input "BANANA" ("BANANA" "Banana" "banana")) (expected) (actual error "sort" "Wrong type argument in position ~A: ~S" 1 #<procedure char<? (#:optional _ _ . _)>) (stdout . "")) (fail (description . "words other than themselves can be anagrams") (code test-success "words other than themselves can be anagrams" (lambda (xs ys) (equal? (list-sort string<? xs) (list-sort string<? ys))) anagram (quote ("LISTEN" ("Listen" "Silent" "LISTEN"))) (quote ("Silent"))) (input "LISTEN" ("Listen" "Silent" "LISTEN")) (expected "Silent") (actual error "sort" "Wrong type argument in position ~A: ~S" 1 #<procedure char<? (#:optional _ _ . _)>) (stdout . "")))
```

**Comment by guygastineau** [comment]
@Pedro-V you will need to update your exercise to the newest version through the exercism web ui.  Sometimes this can take a minute.  It is not entirely clear to me how the updating and notification of available new versions works in the exercism webapp, but you will need to do that to get the updated `test-util.scm`, and then the test runner will run with the new tests.

**Comment by guygastineau** [comment]
After upgrading the exercism you will probably have to `exercism download -t scheme -e anagram` again too.  It won't overwrite your solution code though.  It will just upgrade the `test-util.ss` file but only after upgrading to new exercise version in the ui.

**Comment by guygastineau** [comment]
I just checked out my `anagram` solution in the scheme track.  When I visited the page it had a notification at the top of the screen announcing the new version of the tests available. I clicked the button to upgrade, and then it ran it in the test runner automatically (I didn't have to change any bytes).  So, hopefully you see a notification about a new version being available when you visit the exercise in your browser.
